upstairs/src/test.rs

```diff
@@ -3256,17 +3257,17 @@ mod up_test {
 
         drop(ds);
         // Now, we should be empty, so nw is false
-        assert!(!tokio_test::block_on(up.new_rec_work()).unwrap());
+        assert!(!up.new_rec_work().await.unwrap());
     }
 
-    #[test]
+    #[tokio::test]
     #[should_panic]
-    fn reconcile_leave_no_job_behind() {
+    async fn reconcile_leave_no_job_behind() {
```

Comment:
Ahh... this.

I think we need to make sure that the test is panic'ing in the place we expect it to, instead of in some earlier setup.

And, for this test... I think it's poorly written in that there are several places where it could have panic'd that were not correct.   Maybe let's just file an issue to come back and fix this test specifically, as none of the async changes I think did anything to change the issue that was already there.